### DISCUSSION TITLE
What about Bun runtime support?

### DISCUSSION THREAD

**Discussion by itsfuad** [socketio/socket.io-deno#19] [Q&A] [answered]
Binding with External frameworks doesn't work as there is no io.handler() method.

**Comment by darrachequesne** [comment, marked as answer]
Hi! Bun can be used with the classic `socket.io` package.

**Comment by itsfuad** [comment]
Well I need to use it along with Hono server instance. So I tried
```ts

const app = new Hono();

const io = new Server({
  cors: {
    origin: clienturl,
    methods: ["GET", "POST"],
    credentials: true
  }
});

const handler = io.handler(async (req: Request) => {
  //upgrade to websocket
  return await app.fetch(req) || new Response(null, { status: 404 });
});

serve(handler, { port: port });

```

Now, because Bun uses npm packages and that npm version of socket.io server has no hander() method I am not able to use it on the go. I tried to make a wrapper but I messed up.
